MVCDemos/Controllers/ApiControllers/DataGridRowReorderingController.cs

```diff
@@ -28,32 +28,26 @@ public class DataGridRowReorderingController : ApiController {
             var oldOrderIndex = task.OrderIndex;
 
             JsonConvert.PopulateObject(values, task);
+            var newOrderIndex = task.OrderIndex;
 
             Validate(task);
 
-            _context.Tasks.Remove(task);
-            var sortedTasks = _context.Tasks.OrderBy(t => t.OrderIndex).ToList();
+            if(oldOrderIndex != newOrderIndex) {
+                task.OrderIndex = oldOrderIndex;
+                var sortedTasks = _context.Tasks.OrderBy(t => t.OrderIndex).ToList();
 
-            if(oldOrderIndex != task.OrderIndex) {
-                if(oldOrderIndex < task.OrderIndex) {
-                    for(var i = oldOrderIndex; i < task.OrderIndex; i++) {
-
-                        if(sortedTasks.ElementAt(i).OrderIndex > 0) {
-                            sortedTasks.ElementAt(i).OrderIndex--;
-                        }
-                    }
-                } else if(oldOrderIndex > task.OrderIndex) {
-                    for(var i = task.OrderIndex; i < oldOrderIndex; i++) {
-
-                        if(sortedTasks.ElementAt(i).OrderIndex < sortedTasks.Count()) {
-                            sortedTasks.ElementAt(i).OrderIndex++;
-                        }
-                    }
+                if(oldOrderIndex < newOrderIndex) {
+                    for(var i = oldOrderIndex+1; i <= newOrderIndex; i++) {
+                        sortedTasks.ElementAt(i).OrderIndex--;
```

Comment:
Minor: If sortedTasks is list why we use ElementAt here?
https://stackoverflow.com/questions/5326874/why-would-i-use-enumerable-elementat-versus-the-operator